labs/lab2/regression.test.cpp

```diff
@@ -0,0 +1,94 @@
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+
+#include "doctest.h"
+
+#include <iostream>
+#include <stdexcept>
+
+struct Result {
+  double A;
+  double B;
+};
+
+class Regression {
+  int N_{};
+  double sum_x_{};
+  double sum_y_{};
+  double sum_xy_{};
+  double sum_x2_{};
+
+ public:
+  int size() const { return N_; }
+
+  void add(double x, double y) {
+    ++N_;
+    sum_x_ += x;
+    sum_y_ += y;
+    sum_xy_ += x * y;
+    sum_x2_ += x * x;
+  }
+
+  Result fit() const {
+    if (N_ < 2) {
+      throw std::runtime_error{"Not enough points to run a fit"};
+    }
+    double const d = (N_ * sum_x2_ - sum_x_ * sum_x_);
```

Comment:
```suggestion
    double const d = N_ * sum_x2_ - sum_x_ * sum_x_;
```
Non c'e' ragione di mettere le parentesi